Primitives don't work

```java
class Thing {
  private final String one;
  private final int two;
  private final String three;

  @AssistedInject
  Thing(String one, int two, @Assisted String three) {
    this.one = one;
    this.two = two;
    this.three = three;
  }

  String get() {
    return one + two + three;
  }

  @AssistedInject.Factory
  interface Factory {
    Thing create(String three);
  }
}
```
```
Caused by: java.lang.IllegalArgumentException: invalid type parameter: int
        at com.squareup.javapoet.Util.checkArgument(Util.java:53)
        at com.squareup.javapoet.ParameterizedTypeName.<init>(ParameterizedTypeName.java:51)
        at com.squareup.javapoet.ParameterizedTypeName.<init>(ParameterizedTypeName.java:38)
        at com.squareup.javapoet.ParameterizedTypeName.get(ParameterizedTypeName.java:114)
        at com.squareup.inject.assisted.processor.AssistedInjectionKt.getProviderType(AssistedInjection.kt:101)
        at com.squareup.inject.assisted.processor.AssistedInjectionKt.access$getProviderType$p(AssistedInjection.kt:1)
        at com.squareup.inject.assisted.processor.AssistedInjection.brewJava(AssistedInjection.kt:61)
        at com.squareup.inject.assisted.processor.AssistedInjectProcessor.writeAssistedInject(AssistedInjectProcessor.kt:240)
        at com.squareup.inject.assisted.processor.AssistedInjectProcessor.access$writeAssistedInject(AssistedInjectProcessor.kt:46)
        at com.squareup.inject.assisted.processor.AssistedInjectProcessor$process$3.invoke(AssistedInjectProcessor.kt:65)
        at com.squareup.inject.assisted.processor.AssistedInjectProcessor$process$3.invoke(AssistedInjectProcessor.kt:46)
        at com.squareup.inject.assisted.processor.AssistedInjectProcessor$sam$java_util_function_BiConsumer$0.accept(AssistedInjectProcessor.kt)
        at com.squareup.inject.assisted.processor.AssistedInjectProcessor.process(AssistedInjectProcessor.kt:65)
        at org.gradle.api.internal.tasks.compile.processing.DelegatingProcessor.process(DelegatingProcessor.java:62)
        at org.gradle.api.internal.tasks.compile.processing.NonIncrementalProcessor.process(NonIncrementalProcessor.java:45)
        at jdk.compiler/com.sun.tools.javac.processing.JavacProcessingEnvironment.callProcessor(JavacProcessingEnvironment.java:964)
        at jdk.compiler/com.sun.tools.javac.processing.JavacProcessingEnvironment.discoverAndRunProcs(JavacProcessingEnvironment.java:881)
        at jdk.compiler/com.sun.tools.javac.processing.JavacProcessingEnvironment.access$2100(JavacProcessingEnvironment.java:110)
        at jdk.compiler/com.sun.tools.javac.processing.JavacProcessingEnvironment$Round.run(JavacProcessingEnvironment.java:1202)
        at jdk.compiler/com.sun.tools.javac.processing.JavacProcessingEnvironment.doProcessing(JavacProcessingEnvironment.java:1311)
        at jdk.compiler/com.sun.tools.javac.main.JavaCompiler.processAnnotations(JavaCompiler.java:1250)
        at jdk.compiler/com.sun.tools.javac.main.JavaCompiler.compile(JavaCompiler.java:928)
        at jdk.compiler/com.sun.tools.javac.api.JavacTaskImpl.lambda$doCall$0(JavacTaskImpl.java:100)
        at jdk.compiler/com.sun.tools.javac.api.JavacTaskImpl.handleExceptions(JavacTaskImpl.java:142)
        ... 65 more
```

--- a/assisted-inject-processor/src/main/java/com/squareup/inject/assisted/processor/AssistedInjection.kt
+++ b/assisted-inject-processor/src/main/java/com/squareup/inject/assisted/processor/AssistedInjection.kt
@@ -98,7 +98,7 @@ private val DependencyRequest.providerType: TypeName
     val type = if (key.isProvider) {
       key.type // Do not wrap a Provider inside another Provider.
     } else {
-      ParameterizedTypeName.get(JAVAX_PROVIDER, key.type)
+      ParameterizedTypeName.get(JAVAX_PROVIDER, key.type.box())
     }
     key.qualifier?.let {
       return type.annotated(it)

--- a/assisted-inject-processor/src/test/java/com/squareup/inject/assisted/processor/AssistedInjectProcessorTest.kt
+++ b/assisted-inject-processor/src/test/java/com/squareup/inject/assisted/processor/AssistedInjectProcessorTest.kt
@@ -120,6 +120,54 @@ class AssistedInjectProcessorTest {
         .generatesSources(expected)
   }
 
+  @Test fun primitive() {
+    val input = JavaFileObjects.forSourceString("test.Test", """
+      package test;
+
+      import com.squareup.inject.assisted.Assisted;
+      import com.squareup.inject.assisted.AssistedInject;
+
+      class Test {
+        @AssistedInject
+        Test(long foo, @Assisted String bar) {}
+
+        @AssistedInject.Factory
+        interface Factory {
+          Test create(String bar);
+        }
+      }
+    """)
+
+    val expected = JavaFileObjects.forSourceString("test.Test_AssistedFactory", """
+      package test;
+
+      import java.lang.Long;
+      import java.lang.Override;
+      import java.lang.String;
+      import javax.inject.Inject;
+      import javax.inject.Provider;
+
+      public final class Test_AssistedFactory implements Test.Factory {
+        private final Provider<Long> foo;
+
+        @Inject public Test_AssistedFactory(Provider<Long> foo) {
+          this.foo = foo;
+        }
+
+        @Override public Test create(String bar) {
+          return new Test(foo.get(), bar);
+        }
+      }
+    """)
+
+    assertAbout(javaSource())
+        .that(input)
+        .processedWith(AssistedInjectProcessor())
+        .compilesWithoutError()
+        .and()
+        .generatesSources(expected)
+  }
+
   @Test fun factoryParameterNameIsNotUsedInTheImpl() {
     val input = JavaFileObjects.forSourceString("test.Test", """
       package test;

